test/unit/spec/LoginRouter_spec.js

```diff
@@ -1534,41 +1537,60 @@ function (Okta, Q, Logger, Errors, BrowserFeatures, WidgetUtil, Bundles, config,
       });
 
       itp('Sends the default accept lang header en with API calls if widget is not configured with a language', function () {
-        return setupLanguage({})
+        var success = jasmine.createSpy('successSpy');
+        return setupLanguage({
+          settings: {
+            globalSuccessFn: success
+          }
+        })
           .then(function (test) {
             test.setNextResponse(resSuccess);
             test.router.navigate('');
-            return Expect.waitForPrimaryAuth();
+            return Expect.waitForPrimaryAuth(test);
           })
-          .then(function () {
+          .then(function (test) {
             var form = new PrimaryAuthForm($sandbox);
             expect(form.isPrimaryAuth()).toBe(true);
             form.setUsername('testuser');
             form.setPassword('testpassword');
             form.submit();
             expect($.ajax.calls.mostRecent().args[0].headers['Accept-Language']).toBe('en');
+
+            // Wait for login success
+            test.setNextResponse(resSuccess);
+            return Expect.wait(function () {
+              return success.calls.count() > 0;
```

Comment:
is `success.calls.count()` always 1? (just trying to be a little more explicit)

Reply:
yes there will be only 1 success call

Reply:
then maybe we could do `calls.count === 1` ?